extensions/wikia/ContributionAppreciation/ContributionAppreciationController.class.php

```diff
@@ -49,9 +93,55 @@ public static function onPageHistoryBeforeList() {
 		return true;
 	}
 
+	public static function onBeforePageDisplay( \OutputPage $out, \Skin $skin ) {
+		if ( self::shouldDisplayApprectiation() ) {
+			Wikia::addAssetsToOutput( 'contribution_appreciation_user_js' );
+		}
+
+		return true;
+	}
+
 	private static function shouldDisplayApprectiation() {
 		global $wgUser, $wgLang;
 
 		return $wgUser->isLoggedIn() && $wgLang->getCode() === 'en';
+
+	}
+
+	private function prepareAppreciations( $upvotes ) {
+		$appreciatons = [];
+
+		foreach( $upvotes as $upvote ) {
```

Comment:
This method looks long - can you split the logic inside of this loop into smaller chunks?